### PR TITLE
New topic - show or hide password checkbox implementation

### LINKED ISSUE #8808: New Topic for Show or Hide Password Widget
# New topic request

## Description

Recently, the Show or Hide password checkbox feature is introduced in Magento 2 by the below PR:
PR Link - https://github.com/magento/magento2/pull/31482/

Suggest that rather than create a new topic this might be incorporated into an existing related topic. If not, we can create a new topic on how to use the Show or Hide password checkbox feature in the custom forms.

We can add this new topic like the below UI component document.
https://devdocs.magento.com/guides/v2.4/ui_comp_guide/components/ui-checkbox.html

## Content checklist

-  [x] The topic provides an explanation of how the Show or Hide Password works.
-  [x] The topic provides steps for utilizing the Show or Hide Password in custom forms.
-  [x] The topic contains code samples that show for the Show or Hide Password.
-  [ ] Other (please specify)

## Additional information/resources

- PR Link for Community  Edition - https://github.com/magento/magento2/pull/31482/
- PR Link for Enterprise Edition - https://github.com/magento/partners-magento2ee/pull/451

### PR DESCRIPTION
**Pull request by ajithkumar-maragathavel**
## Purpose of this pull request

This pull request (PR) adds a new topic that describes the implementation of the show or the hide password checkbox functionality.

## Affected DevDocs pages

-  New Topic

## Links to Magento source code

-  https://github.com/magento/magento2/pull/31482

Fixes #8808 

whatsnew
Added the [Show or Hide Password Checkbox](https://devdocs.magento.com/guides/v2.4/ui_comp_guide/howto/show_or_hide_password_checkbox.html) topic.

### CODE REVIEW

File: src/guides/v2.3/ui_comp_guide/howto/show_or_hide_password_checkbox.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+## Example
+
+This example shows the custom implemented company login form template(phtml file) with the show or hide password checkbox.
+
+```phtml
+<div class="block-content" aria-labelledby="block-company-login-heading">
+    <form class="form form-company-login" action="<?= $block->escapeUrl($block->getPostActionUrl()) ?>" method="post" id="company-login-form">
+        <?= $block->getBlockHtml('formkey') ?>
+        <fieldset class="fieldset login" data-hasrequired="<?= $block->escapeHtml(__('* Required Fields')) ?>">
```
Comment (BarnyShergold) [inline src/guides/v2.3/ui_comp_guide/howto/show_or_hide_password_checkbox.md:65]: Block Escaper is deprecated.
```suggestion
        <fieldset class="fieldset login" data-hasrequired="<?= $escaper->escapeHtml(__('* Required Fields')) ?>">
```

File: src/guides/v2.3/ui_comp_guide/howto/show_or_hide_password_checkbox.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+|`component`|The path to the component’s `.js` file in terms of RequireJS. Here the component path value is `Magento_Customer/js/show-password`|String|
+|`passwordSelector`|It must be equals to the id value of the password input field. i.e. `#pass`.|String|
+
+## Example
+
+This example shows the custom implemented company login form template(phtml file) with the show or hide password checkbox.
+
+```phtml
+<div class="block-content" aria-labelledby="block-company-login-heading">
+    <form class="form form-company-login" action="<?= $block->escapeUrl($block->getPostActionUrl()) ?>" method="post" id="company-login-form">
```
Comment (BarnyShergold) [inline src/guides/v2.3/ui_comp_guide/howto/show_or_hide_password_checkbox.md:63]:
```suggestion
    <form class="form form-company-login" action="<?= $escaper->escapeUrl($block->getPostActionUrl()) ?>" method="post" id="company-login-form">
```

File: src/guides/v2.3/ui_comp_guide/howto/show_or_hide_password_checkbox.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+        <?= $block->getBlockHtml('formkey') ?>
+        <fieldset class="fieldset login" data-hasrequired="<?= $block->escapeHtml(__('* Required Fields')) ?>">
+            <div class="field email required">
+                <label class="label" for="email"><span><?= $block->escapeHtml(__('Company Email')) ?></span></label>
```
Comment (BarnyShergold) [inline src/guides/v2.3/ui_comp_guide/howto/show_or_hide_password_checkbox.md:67]:
```suggestion
                <label class="label" for="email"><span><?= $escaper->escapeHtml(__('Company Email')) ?></span></label>
```

File: src/guides/v2.3/ui_comp_guide/howto/show_or_hide_password_checkbox.md [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+                <div class="control">
+                    <input name="login[username]" id="email" type="email" class="input-text" data-validate="{required:true, 'validate-email':true}">
+                </div>
+            </div>
+            <div class="field password required">
+                <label for="pass" class="label"><span><?= $block->escapeHtml(__('Password')) ?></span></label>
```
Comment (BarnyShergold) [inline src/guides/v2.3/ui_comp_guide/howto/show_or_hide_password_checkbox.md:73]:
```suggestion
                <label for="pass" class="label"><span><?= $escaper->escapeHtml(__('Password')) ?></span></label>
```

File: src/guides/v2.3/ui_comp_guide/howto/show_or_hide_password_checkbox.md [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+                <div class="control">
+                    <input name="login[password]" type="password" class="input-text" id="pass" data-validate="{required:true}">
+                </div>
+            </div>
+            <div class="field choice" data-bind="scope: 'showPassword'">
+                <!-- ko template: getTemplate() --><!-- /ko -->
+            </div>
+            <div class="actions-toolbar">
+                <div class="primary">
+                    <button type="submit" class="action login primary" id="company-submit"><span><?= $block->escapeHtml(__('Company Sign In')) ?></span></button>
```
Comment (BarnyShergold) [inline src/guides/v2.3/ui_comp_guide/howto/show_or_hide_password_checkbox.md:83]:
```suggestion
                    <button type="submit" class="action login primary" id="company-submit"><span><?= $escaper->escapeHtml(__('Company Sign In')) ?></span></button>
```

**Review by BarnyShergold — CHANGES_REQUESTED** [review]
Great new topic explanation!

**Comment by ajithkumar-maragathavel** [comment]
@BarnyShergold Thanks for the review. I have updated the PR based on your comments. Please have a look at it.

**Review by BarnyShergold — APPROVED** [review]
Nice!

**Comment by dobooth** [comment]
Nice work, @ajithkumar-maragathavel. Rather than symlink the image, you can put it in `src/common/images` and call it from there in both versions.

**Comment by ajithkumar-maragathavel** [comment]
> Nice work, @ajithkumar-maragathavel. Rather than symlink the image, you can put it in `src/common/images` and call it from there in both versions.

@dobooth  Thanks for the suggestion. I have updated the PR by moving the image to the common directory and removed symlinks as well. Please review it.

**Review by dobooth — APPROVED** [review]

**Comment by dobooth** [comment]
@magento import pr to magento-devdocs/devdocs

**Comment by magento-engcom-team** [comment]
@dobooth the pull request successfully imported.

**Comment by ajithkumar-maragathavel** [comment]
@dobooth I have updated the new topic only for v2.4. Please review it.

**Review by dobooth — APPROVED** [review]

**Comment by dobooth** [comment]
@magento import code to magento-devdocs/devdocs

**Comment by atwixfirster** [comment]
Hi,  ajithkumar-maragathavel !

Thank you for your work here. Sorry for being a bit late on the party but as a reader I would like to know what is a `Magento_Customer/js/show-password` component and where is I can find it in Magento codebase?

Thank you!

**Comment by ajithkumar-maragathavel** [comment]
> Hi, @ajithkumar-maragathavel !
> 
> Thank you for your work here. Sorry for being a bit late on the party but as a reader I would like to know what is a `Magento_Customer/js/show-password` component and where is I can find it in Magento codebase?
> 
> Thank you!

@atwixfirster  Thanks for the suggestion. I will update the document in a separate PR explaining the `Magento_Customer/js/show-password` component and I will also add the Magento codebase link.

**Comment by atwixfirster** [comment]
> > Hi, @ajithkumar-maragathavel !
> > Thank you for your work here. Sorry for being a bit late on the party but as a reader I would like to know what is a `Magento_Customer/js/show-password` component and where is I can find it in Magento codebase?
> > Thank you!
> 
> @atwixfirster Thanks for the suggestion. I will update the document in a separate PR explaining the `Magento_Customer/js/show-password` component and I will also add the Magento codebase link.

Cool. Thank you, @ajithkumar-maragathavel

**Pull request merged**